src/interface_r/R/data_prep.R

```diff
@@ -0,0 +1,102 @@
+#'Prep data for h2o4gpu algorithms
+#'
+#'@param data_table `data.table` object containing data that needs to be preprocessed for h2o4gpu
+#'@param data_path Path to data that needs to be preprocessed for h2o4gpu. Only needed if `data_table` parameter is not provided/available
+#'@param response Response column as a string or index
+#'@param save_as_csv Whether to save processed data as a csv
+#'@param save_csv_path Path to save processed data as a csv
+#'@export
+prep_data <- function(data_table=NULL, data_path=NULL, response=NULL, save_as_csv=FALSE, save_csv_path=NULL){
+  
+  if (save_as_csv && is.null(save_csv_path)) {
+    stop("`save_as_csv` is set to TRUE but `save_csv_path` is NULL. Please specify a path to save processed data")
+  }
+  
+  if (is.null(data_table)) {
+    if (!is.null(data_path)) {
+      print(paste0("Reading in -> ",data_path))
+      DT <- fread(data_path)
+    } else {
+      stop("Parameters `data_table` & `data_path` are NULL. Need to specify at least one.")
+    }
+  } else {
+    DT <- data_table
+  }
+  
+  if (!is.null(response)) {
+    response <- response
+    if (is.character(response)) {
+      print(paste0("Response is -> ",response))
+    } else {
+      print(paste0("Response is -> ",colnames(DT)[response]))
+    }
+  } else {
+    stop("Response is not specified")
+  }
+  
+  print(paste0("Number of columns: ", ncol(DT)))
+  
+  print(paste0("Number of rows: ", nrow(DT)))
+
+  ## Label-encoding of categoricals (those cols with fewer than 1k levels, but not constant)
+  print("Label encoding dataset...")
+  feature.names <- setdiff(names(DT), response)
+  for (ff in feature.names) {
+    tt <- uniqueN(DT[[ff]])
+    if (tt < 1000 && tt > 1) {
+      DT[, (ff):=factor(DT[[ff]])]  
+      print(paste0(ff," has ",tt," levels"))
+    }
+    if (tt < 2) {
+      print(paste0("Dropping constant column: ", ff))
+      DT[, (ff):=NULL]
+    }
+  }
+  
+  print(paste0("Number of columns after label encoding: ", ncol(DT)))
+  
+  numCols <- names(DT)[which(sapply(DT, is.numeric))]
+  catCols <- names(DT)[which(sapply(DT, is.factor))]
+  print(paste0("Number of numeric columns: ",     length(numCols)))
+  print(paste0("Number of categorical columns: ", length(catCols)))
+  
+  ## impute missing values, drop near-const cols and standardize the data
+  print("Imputing missing values using mean...")
```

Comment:
why using mean? I'd imagine users want to do custom imputation so maybe we should expose this somehow

Reply:
for example, they want to pass their own centers and scales or totally different imputation algorithm

Reply:
Yes, we can expose it. This is just a first iteration.